src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -163,6 +164,8 @@ public final boolean nowInMillisUsed() {
 
     public abstract InnerHitsContext innerHits();
 
+    public abstract Set<FieldMapper<?>> aliasFields();
```

Comment:
while I understand that we already have `aliasFilter`... I really think it's a bad name... the alias functionality leaks into the search context (which has nothing to do with aliases). I'd rather call it something like "includeFields"... or something generic like that (the fact that these are coming from an alias is just a use case IMO)